waterbox/uae/Makefile

```diff
@@ -0,0 +1,274 @@
+	$(EMU)/cpuemu_24.c \
+	$(EMU)/cpuemu_31.c \
+	$(EMU)/cpuemu_32.c \
+	$(EMU)/cpuemu_33.c \
+	$(EMU)/cpuemu_34.c \
+	$(EMU)/cpuemu_35.c \
+	$(EMU)/cpuemu_40.c \
+	$(EMU)/cpuemu_50.c \
+	$(EMU)/cpummu.c \
+	$(EMU)/cpummu30.c \
+	$(EMU)/cpustbl.c \
+	$(EMU)/crc32.c \
+	$(EMU)/custom.c \
+	$(EMU)/debug.c \
+	$(EMU)/debugmem.c \
+	$(EMU)/devices.c \
+	$(EMU)/disasm.c \
+	$(EMU)/disk.c \
+	$(EMU)/diskutil.c \
+	$(EMU)/dongle.c \
+	$(EMU)/drawing.c \
+	$(EMU)/driveclick.c \
+	$(EMU)/ethernet.c \
+	$(EMU)/events.c \
+	$(EMU)/expansion.c \
+	$(EMU)/fdi2raw.c\
+	$(EMU)/filesys.c \
+	$(EMU)/filesys_unix.c \
+	$(EMU)/flashrom.c \
+	$(EMU)/fpp.c \
+	$(EMU)/fpp_native.c \
+	$(EMU)/fpp_softfloat.c \
+	$(EMU)/fsdb.c \
+	$(EMU)/fsdb_unix.c \
+	$(EMU)/fsusage.c \
+	$(EMU)/gayle.c \
+	$(EMU)/gfxboard.c \
+	$(EMU)/gfxlib.c \
+	$(EMU)/gfxutil.c \
+	$(EMU)/hardfile.c \
+	$(EMU)/hardfile_unix.c \
+	$(EMU)/hrtmon.rom.c \
+	$(EMU)/ide.c \
+	$(EMU)/idecontrollers.c \
+	$(EMU)/identify.c \
+	$(EMU)/ini.c \
+	$(EMU)/inputdevice.c \
+	$(EMU)/isofs.c \
+	$(EMU)/keybuf.c \
+	$(EMU)/main.c \
+	$(EMU)/memory.c \
+	$(EMU)/misc.c \
+	$(EMU)/missing.c \
+	$(EMU)/native2amiga.c \
+	$(EMU)/ncr_scsi.c \
+	$(EMU)/ncr9x_scsi.c \
+	$(EMU)/newcpu.c \
+	$(EMU)/newcpu_common.c \
+	$(EMU)/pci.c \
+	$(EMU)/picasso96.c \
+	$(EMU)/readcpu.c \
+	$(EMU)/rommgr.c \
+	$(EMU)/rtc.c \
+	$(EMU)/sampler.c \
+	$(EMU)/sana2.c \
+	$(EMU)/savestate.c \
+	$(EMU)/scsi.c \
+	$(EMU)/scsiemul.c \
+	$(EMU)/scsitape.c \
+	$(EMU)/sndboard.c \
+	$(EMU)/specialmonitors.c \
+	$(EMU)/statusline.c \
+	$(EMU)/test_card.c\
+	$(EMU)/traps.c\
+	$(EMU)/uaelib.c \
+	$(EMU)/uaenet.c \
+	$(EMU)/uaeresource.c \
+	$(EMU)/uaeserial.c \
+	$(EMU)/writelog.c \
+	$(EMU)/x86.c \
+	$(EMU)/zfile.c \
+	$(EMU)/zfile_archive.c \
+	$(EMU)/softfloat/softfloat.c \
+	$(EMU)/softfloat/softfloat_decimal.c \
+	$(EMU)/softfloat/softfloat_fpsp.c
+
+SOURCES_C += \
+	$(RETRODEP)/gui.c \
+	$(RETRODEP)/main.c \
+	$(RETRODEP)/mman.c \
+	$(RETRODEP)/parser.c \
+	$(RETRODEP)/serial_host.c \
+	$(RETRODEP)/machdep/support.c \
+	$(RETRODEP)/sounddep/sound.c \
+	$(RETRODEP)/stubs/inputrecord.c \
+	$(RETRODEP)/threaddep/thread.c
+
+SOURCES_C += \
+    $(EMU)/archivers/dms/crc_csum.c \
+	$(EMU)/archivers/dms/getbits.c \
+	$(EMU)/archivers/dms/maketbl.c \
+	$(EMU)/archivers/dms/pfile.c \
+	$(EMU)/archivers/dms/tables.c \
+	$(EMU)/archivers/dms/u_deep.c \
+	$(EMU)/archivers/dms/u_heavy.c \
+	$(EMU)/archivers/dms/u_init.c  \
+	$(EMU)/archivers/dms/u_medium.c \
+	$(EMU)/archivers/dms/u_quick.c \
+	$(EMU)/archivers/dms/u_rle.c \
+	$(EMU)/archivers/lha/crcio.c \
+	$(EMU)/archivers/lha/dhuf.c \
+	$(EMU)/archivers/lha/header.c \
+	$(EMU)/archivers/lha/huf.c \
+	$(EMU)/archivers/lha/larc.c \
+	$(EMU)/archivers/lha/lhamaketbl.c \
+	$(EMU)/archivers/lha/lharc.c \
+	$(EMU)/archivers/lha/shuf.c \
+	$(EMU)/archivers/lha/slide.c \
+	$(EMU)/archivers/lha/uae_lha.c \
+	$(EMU)/archivers/lha/util.c \
+	$(EMU)/archivers/mp2/kjmp2.c
+
+ifeq ($(HAVE_MPEG2), 1)

```

Comment:
this `HAVE_MPEG2`, as well as the `STATIC_LINKING` below don't seem to get set anywhere, so you should probably remove those checks and keep/remove the branches based on the value. It looks like these libmpeg2 sources are not intended to be linked in, but all the below sources are.

Reply:
`HAVE_MPEG2` is used for cd32. Unsure what exactly `STATIC_LINKING` changes in the end result.

Reply:
So is this supposed to be passed to the makefile? I don't see this being set anywhere except for the makeile in `libretro-uae`, which isn't used.

Reply:
I don't plan to have such options. If the feature itself works, I'll just always enable it.